www/perl-lib/SMT/Utils.pm

```diff
@@ -127,6 +129,46 @@ sub db_connect
     return $dbh;
 }
 
+=item hasRegSharing()
+
+Check whether registration sharing is enabled or not.
+
+=cut
+
+sub hasRegSharing
+{
+    my $r = shift;
+
+    if (! defined $REGSHARING) {
+        $REGSHARING = 0;
+        my $cfg;
+        eval
+        {
+            $cfg = SMT::Utils::getSMTConfig();
+        };
+        if($@ || !defined $cfg)
+        {
+            if (! $r)
+            {
+                return -1;
```

Comment:
This "return -1"  will not work, because "-1" is true if check for a bool.
Return undef or 0 to indicate a "false"